File: GOF/Factory Method/java/src/BombedMazeGame.java
import maze.BombedWall;
import maze.Room;
import maze.RoomWithABomb;
import maze.Wall;

public class BombedMazeGame extends MazeGame {
    @Override
    public Wall makeWall() {
        return new BombedWall();
    }

    @Override
    public Room makeRoom(int n) {
        return new RoomWithABomb(n);
    }
}

File: GOF/Factory Method/cpp/Creator/BombedMazeGame.cpp
//
// Created by gaojian on 19-3-7.
//

#include "BombedMazeGame.h"
#include "../maze/bombedwall.h"
#include "../maze/roomwithabomb.h"

using std::make_shared;

std::shared_ptr<Wall> BombedMazeGame::makeWall() const
{
	return make_shared<BombedWall>();
}

std::shared_ptr<Room> BombedMazeGame::makeRoom(int n) const
{
	return make_shared<RoomWithABomb>(n);
}
